### PR TITLE
 allow running without securityContext.procMount=Unmasked (BuildKit v0.4)

### PR DESCRIPTION
**Pull request by AkihiroSuda**
When `/proc` is not mountable, img now automatically disables process sandbox (PID namespace isolation).

Note that this allows build containers to `kill(2)` (and potentially `ptrace(2)` when seccomp is unavailable) the `img` process.

To run `img` in a Docker container, you no longer need to specify `--privileged`, but you still need to specify `--security-opt seccomp=unconfined --security-opt apparmor=unconfined` (which are unconfined on Kubernetes by default).

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@jessfraz @tonistiigi

File: client/workeropt.go [hunk 1]
```diff
@@ -130,3 +133,13 @@ func (c *Client) createWorkerOpt(withExecutor bool) (opt base.WorkerOpt, err err
 
 	return opt, err
 }
+
+func processMode() executoroci.ProcessMode {
+	unshareArgs := []string{"-mfp", "--mount-proc"}
```
Comment (AkihiroSuda) [inline client/workeropt.go]: this doesn't work with Ubuntu 14.04 :(
```
# unshare --help

Usage:
 unshare [options] <program> [args...]

Options:
 -h, --help        usage information (this)
 -m, --mount       unshare mounts namespace
 -u, --uts         unshare UTS namespace (hostname etc)
 -i, --ipc         unshare System V IPC namespace
 -n, --net         unshare network namespace

For more information see unshare(1).
```

**Comment by AkihiroSuda** [comment]
@jessfraz could you take a look?

**Comment by jessfraz** [comment]
Thanks!!!

**Pull request merged**